futures-util/src/future/never_error.rs

```diff
@@ -0,0 +1,39 @@
+use core::marker::Unpin;
+use core::mem::PinMut;
+use futures_core::future::Future;
+use futures_core::task::{self, Poll};
+
+/// Future for the `never_error` combinator, turning a `Future` into a `TryFuture`.
+///
+/// This is created by the `FutureExt::never_error` method.
+#[derive(Debug)]
+#[must_use = "futures do nothing unless polled"]
+pub struct NeverError<Fut> {
+    future: Fut,
+}
+
+impl<Fut> NeverError<Fut> {
+    unsafe_pinned!(future: Fut);
+
+    /// Creates a new NeverError.
+    pub(super) fn new(future: Fut) -> NeverError<Fut> {
+        NeverError { future }
+    }
+}
+
+impl<Fut: Unpin> Unpin for NeverError<Fut> {}
+
+impl<Fut, T> Future for NeverError<Fut>
+    where Fut: Future<Output = T>,
+{
+    type Output = Result<T, ()>;
```

Comment:
I didn't know that [`Never`](https://github.com/rust-lang-nursery/futures-rs/blob/0.2/futures-core/src/never.rs) was introduced only in 0.2 when I proposed the name `NeverError`. I think we should come up with another name for this type. Maybe `UnitError`?

That said, I continue to believe that this combinator is useful. `.map(|x| Ok::<_, ()>(x))` would be damn annoying to type.